### PR TITLE
[DI] Remove AutowireExceptionPass

### PR DESCRIPTION
**Pull request by ogizanagi**
| Q             | A
| ------------- | ---
| Branch?       | master 
| Bug fix?      | no
| New feature?  | yes 
| BC breaks?    | yes
| Deprecations? | no 
| Tests pass?   | yes
| Fixed tickets | #24290 
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php [hunk 1]
```diff
@@ -42,18 +42,6 @@ public function __construct($throwOnAutowireException = true)
...
     /**
-     * @deprecated since version 3.4, to be removed in 4.0.
-     *
-     * @return AutowiringFailedException[]
-     */
-    public function getAutowiringExceptions()
-    {
-        @trigger_error('Calling AutowirePass::getAutowiringExceptions() is deprecated since Symfony 3.4 and will be removed in 4.0. Use Definition::getErrors() instead.', E_USER_DEPRECATED);
-
-        return $this->autowiringExceptions;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php]: the property should be removed also (same in the other pass)

Comment (ogizanagi) [reply src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php]: Good catch. Done.

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Component/DependencyInjection/CHANGELOG.md [hunk 2]
```diff
@@ -22,6 +22,7 @@ CHANGELOG
  * removed support for setting and accessing private services in `Container`
  * removed support for setting pre-defined services in `Container`
  * removed support for case insensitivity of parameter names
+ * removed `AutowireExceptionPass` and `AutowirePass::getAutowiringExceptions()`, use `Definition::addError()` and the `DefinitionErrorExceptionPass` instead
```
Comment (stof) [inline src/Symfony/Component/DependencyInjection/CHANGELOG.md]: Not having anything in the 3.x changelog about the deprecation looks weird to me. The changelog should be updated (in a separate branch)

Comment (ogizanagi) [reply src/Symfony/Component/DependencyInjection/CHANGELOG.md]: See #24399 :)

**Comment by nicolas-grekas** [comment]
can be rebased now that  #24399 is merged.

**Review by dunglas — APPROVED** [review]

**Comment by ogizanagi** [comment]
Rebased

**Comment by nicolas-grekas** [comment]
Thank you @ogizanagi.

**Pull request merged**